internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -550,7 +592,7 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(instance *ai
 			Handler: corev1.Handler{
 				HTTPGet: &corev1.HTTPGetAction{
 					Path:   "/health",
-					Port:   intstr.FromInt(int(servicePort)),
+					Port:   intstr.FromInt(int(serviceHTTPSPort)),
```

Comment:
Just wanted to explicitly point out that, if for whatever reason, `SERVE_HTTPS` is switched to `False` then this will break the health checks.

Reply:
Yeah, I was relying on the fact that this is all hardcoded in the controller.
If we create some field in the CR to indicate how we want to configure the service wrt https then this would all need to be conditional.

Reply:
It's something to keep in mind though as https://github.com/openshift/assisted-service/pull/1593 makes it possible to override.

Reply:
That's an interesting approach.
To solve this we would need to introspect the configmap, which I guess isn't so bad.